src/sdl2_compat.c

```diff
@@ -3914,6 +3914,231 @@ DECLSPEC void SDLCALL SDL_SIMDFree(void *ptr)
+    SDL_zerop(cvt);
+
+#if 0
+    if (!SDL_IsSupportedAudioFormat(src_format)) {
+        return SDL_SetError("Invalid source format");
+    }
+    if (!SDL_IsSupportedAudioFormat(dst_format)) {
+        return SDL_SetError("Invalid destination format");
+    }
+    if (!SDL_IsSupportedChannelCount(src_channels)) {
+        return SDL_SetError("Invalid source channels");
+    }
+    if (!SDL_IsSupportedChannelCount(dst_channels)) {
+        return SDL_SetError("Invalid destination channels");
+    }
+    if (src_rate <= 0) {
+        return SDL_SetError("Source rate is equal to or less than zero");
+    }
+    if (dst_rate <= 0) {
+        return SDL_SetError("Destination rate is equal to or less than zero");
+    }
+    if (src_rate >= SDL_MAX_SINT32 / RESAMPLER_SAMPLES_PER_ZERO_CROSSING) {
+        return SDL_SetError("Source rate is too high");
+    }
+    if (dst_rate >= SDL_MAX_SINT32 / RESAMPLER_SAMPLES_PER_ZERO_CROSSING) {
+        return SDL_SetError("Destination rate is too high");
+    }
+#endif
+
+    stream = SDL3_CreateAudioStream(src_format, src_channels, src_rate,
+            dst_format, dst_channels, dst_rate);
+    if (stream == NULL) {
+        return -1;
+    }
+
+#if DEBUG_CONVERT
+    SDL_Log("SDL_AUDIO_CONVERT: Build format %04x->%04x, channels %u->%u, rate %d->%d\n",
+            src_format, dst_format, src_channels, dst_channels, src_rate, dst_rate);
+#endif
+
+    /* Start off with no conversion necessary */
+    cvt->src_format = src_format;
+    cvt->dst_format = dst_format;
+    cvt->needed = 0;
+    cvt->filter_index = 0;
+    SDL_zeroa(cvt->filters);
+    cvt->len_mult = 1;
+    cvt->len_ratio = 1.0;
+    cvt->rate_incr = ((double)dst_rate) / ((double)src_rate);
+
+    /* Use the filters[] to store some data ... */
+    {
+        AudioParam *ptr = (AudioParam *)&cvt->filters[1];
+        cvt->filters[0] = NULL;
+
+        ptr->src_format = src_format;
+        ptr->src_channels = src_channels;
+        ptr->src_rate = src_rate;
+        ptr->dst_format = dst_format;
+        ptr->dst_channels = dst_channels;
+        ptr->dst_rate = dst_rate;
+
+        cvt->needed = 1;
+        if (src_format == dst_rate && src_rate == dst_rate && src_channels == dst_channels) {
+            cvt->needed = 0;
+        }
+ 
+        if (src_rate < dst_rate) {
+            const int mult = (dst_rate / src_rate);
+            cvt->len_mult *= mult;
+            cvt->len_ratio *= mult;
+        } else {
+            const int div = (src_rate / dst_rate);
+            cvt->len_ratio /= div;
+        }
+       
+        if (src_channels < dst_channels) {
+            cvt->len_mult = ((cvt->len_mult * dst_channels) + (src_channels - 1)) / src_channels;
+        }
+
+        /* !!! FIXME: remove this if we can get the resampler to work in-place again. */
+        /* the buffer is big enough to hold the destination now, but
+           we need it large enough to hold a separate scratch buffer. */
+        cvt->len_mult *= 2;
```

Comment:
Since this is going to push through an SDL_AudioStream that manages its own buffer, we don't need this anymore, since we don't need the app to allocate twice as much space just to make sure we have enough scratch memory to do the work in-place in their buffer.

Reply:
yes removed, (wasn't sure if I could )